From e28634122acb0defa646feee1168b1aa94340b84 Mon Sep 17 00:00:00 2001
From: Vaciller <wyc_19533480830@outlook.com>
Date: Fri, 21 Aug 2026 16:33:25 +0800
Subject: [PATCH] fix: simplify fan cooling levels and enable temp trips on ATK
 board

    - dts: collapse 21-step cooling-levels to a 6-step curve
    - dts: enable rockchip,temp-trips so fan control thresholds apply
---
 arch/arm64/boot/dts/rockchip/rk3588-atk-evb7-lp4.dtsi | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-atk-evb7-lp4.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-atk-evb7-lp4.dtsi
index 84355d277..7b5590579 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-atk-evb7-lp4.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-atk-evb7-lp4.dtsi
@@ -66,12 +66,7 @@ fan: pwm-fan {
 		compatible = "pwm-fan";
 		#cooling-cells = <2>;
 		pwms = <&pwm3 0 50000 0>;
-		cooling-levels = <
-			0 50 100 120 150 170 180 200
-			205 210 215 220 225 230 235 240
-			245 250 251 252 253 254 255
-		>;
-		/*
+		cooling-levels = <0 50 100 150 200 255>;
 		rockchip,temp-trips = <
 			50000	1
 			55000	2
@@ -79,7 +74,6 @@ fan: pwm-fan {
 			65000	4
 			70000	5
 		>;
-		*/
 	};
 
 	hdmiin-sound {
-- 
2.53.0

